### PR TITLE
Add syntax highlighting to nested code blocks #6656

### LINKED ISSUE #6656: Add syntax highlighting to nested code blocks

### PR DESCRIPTION
**Pull request by ashutuptiwari**
## Checklist

- [ ] **Page addition**
  - [ ] Add `permalink` to pages.

- [ ] **Page removal**
  - [ ] Remove entry from corresponding left nav YAML file.
  - [ ] Add `redirect_from` on the alternative page.
  - [ ] If no alternative page, add redirection on the `redirections.js ` file.

- [ ] **Page rename**
  - [ ] Add front-matter `redirect_from`.
  - [ ] Add front-matter `redirect_to:` (if applicable).

- [ ] **Page restrcuture**
  - [ ] Add `permalink` to pages.
  - [ ] Add front-matter `redirect_from`.
  - [ ] Add front-matter `redirect_to:` (if applicable).

### CODE REVIEW

**Review by sm1990 — CHANGES_REQUESTED** [review]
@ashutuptiwari Thank you for your contribution.

However, your fix breaks for normal code snippets unless we add a space after every ending `. Also if we're accepting this solution, we have to update all our md files and add the space, which is not practical. We need to look for a different method.
image

**Comment by ashutuptiwari** [comment]
Alright let me try something else

File: swan-lake/development-tutorials/test-document-the-code/generate-code-documentation.md [hunk 1]
```diff
@@ -172,7 +172,7 @@ Now, let's add a function to the `math` module to be documented. Copy and paste
 public isolated function pow(float a, float b) returns float {
     return 0;
 }
-```
+```
```
Comment (sm1990) [inline swan-lake/development-tutorials/test-document-the-code/generate-code-documentation.md:175]: Remove the space

File: utils/highlighter.js [hunk 2]
```diff
@@ -33,9 +33,10 @@ const highlight = async (content) => {
     });
 
     let codes = new Map();
-    const regex = /```(\w+)([\s\S]*?)```/g;
+    const regex = /```(\w+)([\s\S]*?)\n```/g;
     let match = [];
     while (match = regex.exec(content)) {
+        console.log(`match:\n  ${match[0]}`);
```
Comment (sm1990) [inline utils/highlighter.js:39]: Remove debug logs.

**Comment by ashutuptiwari** [comment]
Done @sm1990

**Comment by sm1990** [comment]
> Done @sm1990

The fix does seem to work for code snippets within `<li>`s. Test http://localhost:3000/learn/integration-tutorials/sending-a-message-to-a-service/
image

**Comment by ashutuptiwari** [comment]
fixed that as well @sm1990

**Review by sm1990 — APPROVED** [review]

**Comment by sm1990** [comment]
@ashutuptiwari Thank you for your contribution. We are currently hosting a `Ballerina Hacktoberfest`. Since this issue is eligible for our event, please visit https://ballerina.io/hacktoberfest/#rules & submit the form. 😄

**Comment by ashutuptiwari** [comment]
You're welcome @sm1990 and thank you for your inputs. This was a great learning opportunity. I look forward to contributing more in the future. 
Hasta La Vista

**Pull request merged**